core/src/main/java/com/linecorp/armeria/internal/DefaultAttributeMap.java

```diff
@@ -70,18 +73,35 @@
     private static final int BUCKET_SIZE = 8;
     private static final int MASK = BUCKET_SIZE  - 1;
 
+    @Nullable
+    private final AttributeMap parentMap;
```

Comment:
I'm trying to think of a word other than `parent` - the word almost always implies a chain of descendants, not just one as we have here (and have to explicitly state in comments). `root` is much more explicit and maybe clear enough? Or maybe `owner`, `creator`.

Not a big deal but want to bring up that `parent` here makes things harder to reason about.

Reply:
`root` sounds good. Thanks!